/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
import "google/protobuf/struct.proto";
package fedlearner_webconsole.proto;

message SystemVariables {
  repeated SystemVariable variables = 1;
}

message SystemVariable {
  string name = 1;
  google.protobuf.Value value = 2;

  enum ValueType{
    STRING = 0;
    INT = 1;
    LIST = 2;
    OBJECT = 3;
  }

  ValueType value_type = 4;
  // When the SystemVariable's fixed is True, user can't delete the Variable or edit its name.
  bool fixed = 5;
}

message SystemInfo {
  // common name parsed from x.509 certificate, e.g. fl-test.com
  string domain_name = 1;
  string name = 2;
  // pure domain name parsed from common name, e.g. test
  string pure_domain_name = 3;
}

message SettingPb {
  string uniq_key = 1;
  string value = 2;
}

message DashboardInformation {
  string name = 1;
  string uuid = 2;
  string url = 3;
}
